tecs/tecs.cpp

```diff
@@ -323,7 +323,7 @@ void TECS::_update_throttle_setpoint(const float throttle_cruise, const matrix::
 		// Assume induced drag scales linearly with normal load factor.
 		// The additional normal load factor is given by (1/cos(bank angle) - 1)
 		float cosPhi = sqrtf((rotMat(0, 1) * rotMat(0, 1)) + (rotMat(1, 1) * rotMat(1, 1)));
-		STE_rate_setpoint = STE_rate_setpoint + _load_factor_correction * (1.0f / constrain(cosPhi, 0.1f, 1.0f) - 1.0f);
+		STE_rate_setpoint = STE_rate_setpoint + _load_factor_correction * (1.0f / constrain(cosPhi * cosPhi, 0.1f, 1.0f) - 1.0f);
```

Comment:
Does it make sense to take the sqrt above if we are just going to be squaring it here anyway?

Reply:
@CarlOlsson Can you also please verify the function for calculating the cosine of roll angle?
I plugged in the values into the formula and I end up with:
`cos(roll)^2 + sin(roll)^2 * sin(pitch)^2`
This is only exact if the pitch angle is zero which is probably not generally a bad assumption.
However, I prefer if we use the matrix lib to calculate the roll angle and then just take the cosine of that. Otherwise we lose the benefit of the test coverage.

Could you also please update the comment above the formula you changed, I don't think it reflects the calculations anymore, does it?

Reply:
@CarlOlsson I assume the rationale behind this is the quadratic shape of Cd as a function of Cl?

Reply:
There are two considerations here:

1) Induced drag scales with CL^2 and is more logically scaled using  load factor^2
2) Roll angle from a Tait-Bryan 321 sequence (what the library returns) is not useful for this type of compensaton because it gives incorrect results for normal load factor at large pitch angles. I think roll from a Tait-Bryan 312 (Yaw->Roll->Pitch) sequence would be more useful.

I will have a look at this and get back to you tomorrow.

Reply:
The roll from a Tait-Bryan 312 sequnece is = asinf(_R_to_earth(2, 1)) 

See the Ekf::fuseHeading() function in the EKF for an example of where this rotation definition is used.

It is possible the exisiting code in TECS is trying to do a pitch compensation, but I think the Roll from a 312 sequence would be more understandable.

Reply:
Whatever method is used should be tested to correctly calculate the load factor for climbing and descending turns.

Reply:
@priseborough makes sense

Reply:
![20190614_082053](https://user-images.githubusercontent.com/3596952/59471249-e8d77000-8e7d-11e9-9dba-9631a766d907.jpg)

See attached for derivation of pitch angle compensation where ignoring angle of attack, pitch angle is defined as angle of X axis above horizontal.

Reply:
Therefore the load factor calculated  both centripetal, forwards and vertical acceleration components would be:

cos(pitch) / (cos(roll)*(1 + sin^2(pitch))

where pitch is angle of X axis above horizontal plane (second rotation angle from Tait-Bryan 321 sequence) and roll angle is angle of Y axis relative to horizontal plane (second rotation angle from Tait-Bryan 312 sequence)